src/app/public/resolves/public-tags-resolve.service.ts

```diff
@@ -0,0 +1,25 @@
+import {
+  ActivatedRouteSnapshot,
+  Resolve,
+  RouterStateSnapshot,
+} from '@angular/router';
+import { ApiService } from '@shared/services/api/api.service';
+import { Injectable } from '@angular/core';
+import { ArchiveVO } from '@models/index';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class PublicTagsResolveService implements Resolve<any> {
+  constructor(private api: ApiService) {}
+
+  async resolve(route: ActivatedRouteSnapshot, state: RouterStateSnapshot) {
+    const archiveNbr = route.params.publicArchiveNbr;
+    const response = await this.api.archive.get([
+      new ArchiveVO({ archiveNbr }),
+    ]);
+    const archiveId = response.getArchiveVO().archiveId;
+    const res = await this.api.archive.getArchiveTags(archiveId);
+    localStorage.setItem('tags', JSON.stringify(res));
```

Comment:
Though this is a bit nitpicky, can you use the `StorageService` to access localStorage instead of accessing it directly? Just want to make sure that all calls to localStorage are consistent!

Also while we're on this topic, I know that the ticket says to store them in localStorage, and that's fine, but would sessionStorage make more sense for storing these tags? (I'm actually not sure since we're only storing them right now, but I try to default to sessionStorage unless I need to use localStorage)

Reply:
I think session storage would be fine, mainly because the API call is being made each time a public archive is visited, so they would get overwritten on each visit anyway